scripts/migrations/racks_plugin.php

```diff
@@ -0,0 +1,461 @@
+
+include ('../../inc/includes.php');
+
+$out = "";
+getArgs();
+checkPluginTables();
+TruncateCoreTables();
+migratePlugin();
+printOutput();
+exit (0);
+
+function getArgs() {
+   if (isset($_SERVER['argv'])) {
+      for ($i=1; $i<$_SERVER['argc']; $i++) {
+         $it    = explode("=", $_SERVER['argv'][$i], 2);
+         $it[0] = preg_replace('/^--/', '', $it[0]);
+
+         $_GET[$it[0]] = (isset($it[1]) ? $it[1] : true);
+      }
+   }
+
+   if ((isset($_SERVER['argv']) && in_array('help', $_SERVER['argv']))
+       || isset($_GET['help'])) {
+      echo "Usage: php -q -f racks_plugin.php\n";
+      echo "skip_error: Don't exit on import error\n";
+      echo "truncate: Remove existing core data\n";
+      exit (0);
+   }
+}
+
+function printOutput() {
+   global $out;
+
+   if (!isCommandLine()) {
+      $out = nlbr($out);
+   }
+   echo $out;
+   $out = "";
+}
+
+function checkError($error = false) {
+   printOutput();
+   if ($error
+       && (!isset($_GET['skip_error']) || !$_GET['skip_error'])) {
+      echo textRed("\nSome errors triggered, aborting!\n");
+      if (count($_SESSION['MESSAGE_AFTER_REDIRECT'])) {
+         var_dump($_SESSION['MESSAGE_AFTER_REDIRECT']);
+      }
+      exit (1);
+   }
+}
+
+function checkResult($result = 0, $print_result = false) {
+   global $out;
+
+   $error = false;
+
+   if ($result !== false && $result > 0) {
+      $out.= textGreen("OK");
+      if ($print_result) {
+         $out.= " (".$result.")";
+      }
+   } else {
+      $out.= textRed("KO");
+      if ($print_result) {
+         if ($result === false) {
+            $result = "false";
+         }
+         $out.= " (".$result.")";
+      }
+      $error = true;
+   }
+   $out.= "\n";
+   printOutput();
+
+   return $error;
+}
+
+function textGreen($text = "") {
+   if (isCommandLine()) {
+      return "\033[32m$text\033[0m";
+   } else {
+      return "<span style='color=green;'>$text</span>";
+   }
+}
+
+function textRed($text = "") {
+   if (isCommandLine()) {
+      return "\033[31m$text\033[0m";
+   } else {
+      return "<span style='color=red;'>$text</span>";
+   }
+}
+
+function textYellow($text = "") {
+   if (isCommandLine()) {
+      return "\033[33m$text\033[0m";
+   } else {
+      return "<span style='color=yellow;'>$text</span>";
+   }
+}
+
+function checkPluginTables() {
+   global $DB, $out;
+
+   $error = false;
+   $out .= "\n## Check presence of racks plugin tables\n\n";
+   $rack_tables = [
+      'glpi_plugin_racks_itemspecifications',
+      'glpi_plugin_racks_rackmodels',
+      'glpi_plugin_racks_racktypes',
+      'glpi_plugin_racks_rackstates',
+      'glpi_plugin_racks_roomlocations',
+      'glpi_plugin_racks_racks',
+      'glpi_plugin_racks_racks_items',
+   ];
+   foreach ($rack_tables as $table) {
+      $out.= "- $table: ";
+      $exist = $DB->tableExists($table);
+      $error = checkResult($exist);
+   }
+
+   checkError($error);
+
+   return $error;
+}
+
+function TruncateCoreTables() {
```

Comment:
Is this really required? I guess adding the possibility as an argument would be better.